scripts/travisnpmpublish.js

```diff
@@ -0,0 +1,28 @@
+const child_process = require('child_process');
+const fs = require('fs');
+
+if (
+  process.env.TRAVIS_BRANCH != 'master' ||
+  (process.env.TRAVIS_PULL_REQUEST && process.env.TRAVIS_PULL_REQUEST != 'false')
+) {
+  console.log('Skipping travis npm publish');
+} else {
+  child_process.exec('git rev-parse --short HEAD', (err, stdout, stderr) => {
```

Comment:
Maybe add some logging for start, finish, and error cases.

Reply:
Errors should be logged by node because of the `throw`s but yes, I will add some more logging during the execution.